### ISSUE TITLE
Bug: Meter: Chambre Internal Server Error, An unexpected error on the SmartThings servers has occurred. These errors should be rare, statusCode: 500

### ISSUE THREAD

**Issue by frankdestclaude** [OpenWonderLabs/homebridge-switchbot#947]
### Describe the Bug

Sur raspberry PI4b, homebridge, error message : Meter: Chambre Internal Server Error, An unexpected error on the SmartThings servers has occurred. These errors should be rare, statusCode: 500

### To Reproduce

Sur raspberry PI4b, homebridge, error message : Meter: Chambre Internal Server Error, An unexpected error on the SmartThings servers has occurred. These errors should be rare, statusCode: 500

### Expected Behavior

qu'iln y ait plus de message d'erreur

### Relevant Log Output

```shell
[29/02/2024 13:18:15] [SwitchBot] Meter: Chambre Internal Server Error, An unexpected error on the SmartThings servers has occurred. These errors should be rare, statusCode: 500
[29/02/2024 13:18:15] [SwitchBot] Meter: Chambre Unknown statusCode: undefined, Submit Bugs Here: ' + 'https://tinyurl.com/SwitchBotBug
[29/02/2024 13:48:16] [SwitchBot] Meter: Chambre Internal Server Error, An unexpected error on the SmartThings servers has occurred. These errors should be rare, statusCode: 500
[29/02/2024 13:48:16] [SwitchBot] Meter: Chambre Unknown statusCode: undefined, Submit Bugs Here: ' + 'https://tinyurl.com/SwitchBotBug
[29/02/2024 13:54:16] [SwitchBot] Meter: Chambre Internal Server Error, An unexpected error on the SmartThings servers has occurred. These errors should be rare, statusCode: 500
[29/02/2024 13:54:16] [SwitchBot] Meter: Chambre Unknown statusCode: undefined, Submit Bugs Here: ' + 'https://tinyurl.com/SwitchBotBug
[29/02/2024 14:26:16] [SwitchBot] Meter: Chambre Internal Server Error, An unexpected error on the SmartThings servers has occurred. These errors should be rare, statusCode: 500
[29/02/2024 14:26:16] [SwitchBot] Meter: Chambre Unknown statusCode: undefined, Submit Bugs Here: ' + 'https://tinyurl.com/SwitchBotBug
[29/02/2024 15:32:16] [SwitchBot] Meter: Chambre Internal Server Error, An unexpected error on the SmartThings servers has occurred. These errors should be rare, statusCode: 500
[29/02/2024 15:32:16] [SwitchBot] Meter: Chambre Unknown statusCode: undefined, Submit Bugs Here: ' + 'https://tinyurl.com/SwitchBotBug
[29/02/2024 15:38:16] [SwitchBot] Meter: Chambre Internal Server Error, An unexpected error on the SmartThings servers has occurred. These errors should be rare, statusCode: 500
[29/02/2024 15:38:16] [SwitchBot] Meter: Chambre Unknown statusCode: undefined, Submit Bugs Here: ' + 'https://tinyurl.com/SwitchBotBug
```

### Config for homebridge-switchbot

```shell
{
    "name": "SwitchBot",
    "credentials": {
        "token": "401b11fdbe36201ea778768176b8a533061ad660e1e6c68947b344c1870ce08bb4dd233f9bcc914b7f3a7cc135e197b7",
        "secret": "2d0baa589a7aa80dd1c57eee360ba625",
        "notice": "Keep your Token & Secret a secret!"
    },
    "platform": "SwitchBot",
    "_bridge": {
        "username": "0E:75:7E:AC:71:C3",
        "port": 48999
    }
}
```

### Device and Model

Raspberry PI4b

### Node.js Version

v20.11.1 

### NPM Version

v10.2.4

### Homebridge Version

v1.7.0

### Homebridge Switchbot Plugin Version

v3.4.0 

### Homebridge Config UI X Plugin Version

v1.7.0

### Operating System

raspbian

**Comment by sametguzeldev** [comment]
I am having the same issue myself.
``` 
[3/12/2024, 5:40:52 AM] [SwitchBot] Bot: Upper Air conditioner Unknown statusCode: undefined, Submit Bugs Here: ' + 'https://tinyurl.com/SwitchBotBug
[3/12/2024, 5:40:52 AM] [SwitchBot] Meter: Bedroom Meter Internal Server Error, An unexpected error on the SmartThings servers has occurred. These errors should be rare, statusCode: 500
[3/12/2024, 5:40:52 AM] [SwitchBot] Meter: Bedroom Meter Unknown statusCode: undefined, Submit Bugs Here: ' + 'https://tinyurl.com/SwitchBotBug
[3/12/2024, 5:40:52 AM] [SwitchBot] Contact Sensor: Garden Door Sensor Internal Server Error,   An unexpected error on the SmartThings servers has occurred. These errors should be rare, statusCode: 500
[3/12/2024, 5:40:52 AM] [SwitchBot] Contact Sensor: Garden Door Sensor Unknown statusCode: undefined, Submit Bugs Here: ' + 'https://tinyurl.com/SwitchBotBug
[3/12/2024, 5:42:52 AM] [SwitchBot] Smart Lock: Main Door Internal Server Error,        An unexpected error on the SmartThings servers has occurred. These errors should be rare, statusCode: 500
[3/12/2024, 5:42:52 AM] [SwitchBot] Smart Lock: Main Door Unknown statusCode: undefined, Submit Bugs Here: ' + 'https://tinyurl.com/SwitchBotBug
[3/12/2024, 5:42:52 AM] [SwitchBot] Meter: Gaming meter Internal Server Error,  An unexpected error on the SmartThings servers has occurred. These errors should be rare, statusCode: 500
[3/12/2024, 5:42:52 AM] [SwitchBot] Meter: Gaming meter Unknown statusCode: undefined, Submit Bugs Here: ' + 'https://tinyurl.com/SwitchBotBug
[3/12/2024, 5:44:52 AM] [SwitchBot] Meter: Gaming meter Internal Server Error,  An unexpected error on the SmartThings servers has occurred. These errors should be rare, statusCode: 500
[3/12/2024, 5:44:52 AM] [SwitchBot] Meter: Gaming meter Unknown statusCode: undefined, Submit Bugs Here: ' + 'https://tinyurl.com/SwitchBotBug
[3/12/2024, 5:44:52 AM] [SwitchBot] Bot: Computer Internal Server Error,        An unexpected error on the SmartThings servers has occurred. These errors should be rare, statusCode: 500
[3/12/2024, 5:44:52 AM] [SwitchBot] Bot: Computer Unknown statusCode: undefined, Submit Bugs Here: ' + 'https://tinyurl.com/SwitchBotBug
[3/12/2024, 5:44:52 AM] [SwitchBot] Contact Sensor: Main Door Contact Sensor Internal Server Error,     An unexpected error on the SmartThings servers has occurred. These errors should be rare, statusCode: 500
```

**Comment by epheterson** [comment]
Same 

https://github.com/OpenWonderLabs/homebridge-switchbot/issues/953

**Comment by niqbert** [comment]
Same

**Comment by Gus-smarthome** [comment]
same

**Comment by zx81uk** [comment]
Same

**Comment by mrlightison** [comment]
Same, for the Smart Lock.

**Comment by Aeon-Auriga** [comment]
Same.

**Comment by swshong** [comment]
Same

**Comment by ghost** [comment]
Same here.

**Comment by nnhoang** [comment]
Same issue as #959 . Reverting to v3.1.3 worked for me.

**Comment by sametguzeldev** [comment]
This is an issue directly with Switchbot API v1.1. I've tested it myself on Postman and I saw the same 500 status code.
![image](https://github.com/OpenWonderLabs/homebridge-switchbot/assets/6827857/c1479f44-5e24-4120-9881-8fae667ebf04)

I did the same test on Switchbot API v1.0 and that seemed to be working. Haven't received any 500 status code but I did not test it that much.
What I saw was if you try to do the request again after a while it succeeds. So I implemented a retry mechanism myself and it worked. Received status code 100 after several tries.

![image](https://github.com/OpenWonderLabs/homebridge-switchbot/assets/6827857/12f8be73-7b8f-4122-af74-fb844f6355f8)
